palace/models/timeoperator.cpp

```diff
@@ -225,24 +358,78 @@ double TimeOperator::GetMaxTimeStep() const
   return 2.0 / std::sqrt(lam);
 }
 
-void TimeOperator::Init()
+void TimeOperator::Init(double &dt)
 {
   // Always use zero initial conditions.
-  E = 0.0;
-  dE = 0.0;
+  sol = 0.0;
   B = 0.0;
-  ode->Init(*op);
+  if (use_mfem_integrator)
+  {
+    ode->Init(*op);
+  }
+#if defined(MFEM_USE_SUNDIALS)
+  if (mfem::ARKStepSolver *arkode = dynamic_cast<mfem::ARKStepSolver *>(ode.get()))
+  {
+    if (!adapt_dt)
+    {
+      // Disable adaptive time stepping.
+      arkode->SetFixedStep(dt);
+    }
+  }
+#endif
 }
 
 void TimeOperator::Step(double &t, double &dt)
 {
-  // Single time step for E-field.
   En = E;
-  ode->Step(E, dE, t, dt);
+  double dt_input = dt;
+  ode->Step(sol, t, dt);
+  // Ensure user-specified dt does not change.
+  dt = dt_input;
 
   // Trapezoidal integration for B-field: dB/dt = -∇ x E.
   En += E;
   Curl->AddMult(En, B, -0.5 * dt);
```

Comment:
I don't think so. We solve the ODE system for E and Edot, not sure how else to get B.

Reply:
FYI you can, but you have to use the (more standard?) `[E, B]` linearization instead of the `[E, dE/dt]` one. This is described in Zhu and Cangellaris and should permit the same linear system for `E` after 2x2 block elimination.

Reply:
I meant no easy way in this `[E, dE/dt]` linearization. I actually tried the `[E, B]` formulation first but I was seeing some weird things like small but persistent solution differences compared the second-order ODE formulation, error vs dt trends not matching the expected order of accuracy, and some instability issues on complex cases. Not sure if it was an issue with the `MixedVectorWeakCurl` operator, some BC consideration (?), or most likely me doing something wrong. We're thinking of moving forward with the `[E, dE/dt]` formulation for now but I will try to revisit this later.